src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -1922,6 +1927,68 @@ mongoc_collection_insert_many (mongoc_collection_t *collection,
                                        error,
                                        "insertedCount");
 
+   // Builds document of insertedIds.
```

Comment:
Suggest reducing the scope of this PR to only report the ID in `mongoc_collection_insert_one`. I expect reporting IDs for multi-document inserts may require significant additional effort.

# Rationale

I expect scanning and copying IDs for large inserts may impact performance.

Only reporting IDs of successful inserts seems complicated for multi-document inserts. The current strategy (removing IDs included in `writeErrors`) may not guarantee that all returned IDs have been acknowledged by the server. Example: if a network error occurs, no `writeErrors` are returned in the reply. Here is a failing test showing this behavior by using a server failpoint to simulate a network error: 

```c
void
test_insert_omits_id_on_network_error (void)
{
   mongoc_client_t *client = test_framework_new_default_client ();
   mongoc_collection_t *coll = get_test_collection (client, "test_insert_omits_id_on_network_error");
   bson_error_t error;
   bool ok;

   // Configure a server failpoint to test a network error.
   {
      // Fail twice. The default behavior retries writes once.
      bool ret = mongoc_client_command_simple (client,
                                               "admin",
                                               tmp_bson (BSON_STR ({
                                                  "configureFailPoint" : "failCommand",
                                                  "mode" : {"times" : 2},
                                                  "data" : {"failCommands" : ["insert"], "closeConnection" : true}
                                               })),
                                               NULL,
                                               NULL,
                                               &error);
      ASSERT_OR_PRINT (ret, error);
   }

   // Test inserting multiple documents omits IDs on a network error.
   {
      bson_t *docs[2] = {tmp_bson ("{'foo1': 'bar1'}"), tmp_bson ("{'foo2': 'bar2'}")};
      bson_t reply;

      ok = mongoc_collection_insert_many (coll, (const bson_t **) docs, 2, NULL /* opts */, &reply, &error);
      ASSERT_OR_PRINT (ok, error);
      // Check that `reply` does not contain any insertedIds:
      ASSERT_MATCH (&reply, "{ 'insertedIds': { '0' : {'$exists' : false}, '1': { '$exists' : false } } }"); // Fails!
      bson_destroy (&reply);
   }

   mongoc_collection_destroy (coll);
   mongoc_client_destroy (client);
}
```

Reporting may be further complicated with batch splitting (e.g. a multi-document write may have a network error in one batch, but successfully insert in another).

I expect these concerns could be further investigated and addressed. However, the description and case attached to CDRIVER-3093 only reference single-document inserts. The linked PHPLIB-428 requests multi-document insert IDs, but is motivated only by testing. I prefer proceeding with only reporting IDs on single-document inserts, and awaiting future motivation for multi-document inserts.